contributor-docs/adrs/adr-020-live-regions.md

```diff
@@ -0,0 +1,233 @@
+# ADR 020: Live Regions
+
+## Status
+
+| Stage    | Status      |
+| -------- | ----------- |
+| Approved | <!-- ✅ --> |
+| Adopted  | <!-- 🚧 --> |
+
+## Context
+
+There are several components in Primer React that make use of live regions.
+However, the library does not have a central pattern for making accessible live
+region announcements. This ADR documents the decision to use a central pattern
+for live region announcements that can be used across Primer and GitHub.
+
+ARIA Live regions fall into several buckets that we may want to use as component
+authors:
+
+- [ARIA live region roles](https://www.w3.org/TR/wai-aria-1.2/#live_region_roles), such as `alert`, `log`, or `status`
+- [The `aria-live` attribute](https://www.w3.org/TR/wai-aria-1.2/#aria-live) to turn an element into a live region
+
+In components, we see the following scenarios in Primer React:
+
+- Announce the contents of an element when it is rendered or on page load, like when a spinner is displayed or a form is submitted
+- Announce the changes to the content of an element, like when the count inside of a
+  button is incremented
+- Announce a message programmatically, such as the number of results for a query
+
+Currently, contributors may reach for roles such as `alert` or `status` to
+achieve these scenarios. They may also add `aria-live="assertive"` or `aria-live="polite"` with an an `aria-atomic="true"` to an element explicitly.
+However, both of these approaches do not announce consistently across screen
+readers. This could be due to live regions being injected dynamically into the document (this includes loading content into the document via React), dynamically changing the visibility of a live region, or some other technique causing an announcement to not be announced.
+
+For more information about the ways in which live regions may not work as
+expected, visit: [Why are my live regions not working?](https://tetralogical.com/blog/2024/05/01/why-are-my-live-regions-not-working/)
+
+### Links & Resources for ARIA Live regions
+
+- https://www.sarasoueidan.com/blog/accessible-notifications-with-aria-live-regions-part-1/
+- https://www.scottohara.me/blog/2022/02/05/are-we-live.html
+
+## Decision
+
+In order to have a common interop point for live region announcements, Primer React will
+make use of a `live-region` custom element from `@primer/live-region-element`.
+This package will be included and published from the `primer/react` repo.
```

Comment:
So we're getting rid of `@primer/live-region-element` and moving it to this repo?

Reply:
Basically this would mean that it wouldn't live in it's own repo and instead would be under `packages/live-region-element` as another package in the monorepo 👀 It would still be published as the `@primer/live-region-element` package.

Reply:
Does this mean that PVC consumers could also pull this element into their projects?

Reply:
Yeah! Definitely @khiga8